system/lib/pthread/pthread_self_stub.c

```diff
@@ -28,4 +29,7 @@ __attribute__((constructor))
 static void init_pthread_self(void) {
   __main_pthread.locale = &libc.global_locale;
   __main_pthread.tid = getpid();
+  __main_pthread.stack = (void*)emscripten_stack_get_base();
+  __main_pthread.stack_size = emscripten_stack_get_base();
```

Comment:
Should this be `emscripten_stack_get_end() - emscripten_stack_get_base()`

Reply:
Indeed...

Reply:
Can you see if using the linker generated symbol `&__stack_high`  and `&__stack_high - &__stack_low` help with the regression?   (Along with `extern int __stack_high; extern int __stack_low;`)

Reply:
Using:
```
-  __main_pthread.stack = (void*)emscripten_stack_get_base();
-  __main_pthread.stack_size = emscripten_stack_get_base() - emscripten_stack_get_end();
+  __main_pthread.stack = (void*)&__stack_low;
+  __main_pthread.stack_size = &__stack_high - &__stack_low;
```
I get:
```
-stack_addr: 0, stack_size: 65536, guard_size: 8192
+stack_addr: 0xffffc000, stack_size: 16384, guard_size: 8192
```
So it seems slightly different? Looking at the code I can't understand why.

Reply:
Also, for `test_pthread_stub`, when I use `emscripten_stack_get_base` and `emscripten_stack_get_end` it comes out 9 bytes smaller...

Reply:
That seems odd, how did you declare `__stack_low` and `__stack_high`?

Reply:
Indeed, with `-Os` the `&__stack_high - &__stack_low;` code comes out 22 bytes smaller.

Reply:
Okay got it working with these.